eng/pipelines/templates/steps/test-packages.yml

```diff
@@ -0,0 +1,111 @@
+parameters:
+  - name: ServiceDirectory
+    type: string
+    default: auto
+  - name: Toolchain
+    type: string
+    default: stable
+  - name: ArtifactSuffix
+    type: string
+    default: "linux"
+  - name: PublishArtifacts
+    type: boolean
+    default: false
+  - name: UnitTests
+    type: boolean
+    default: false
+  - name: FunctionalTests
+    type: boolean
+    default: false
+  - name: TestTimeoutInMinutes
+    type: number
+    default: 60
+
+steps:
+  # using -Zscript with verify-dependencies.rs requires nightly be installed
+  - ${{ if ne(parameters.Toolchain, 'nightly') }}:
+      - template: /eng/pipelines/templates/steps/use-rust.yml@self
+        parameters:
+          Toolchain: nightly
```

Comment:
Is the result of this that we'll install nightly toolchain into every build? I don't think that's necessary. Just running this script *in* nightly on one agent should be sufficient. It just makes sure that all dependencies are centrally managed in the root `Cargo.toml` much like .NET's `eng/Packages.Data.props`.

Reply:
I made the call in test-packages dependent on `$toolchain -eq 'nightly'` rather than installing nightly on every leg